### ISSUE TITLE
fix: granite4_vision text-only forward returns NaN logits ~1 run in 6

### ISSUE THREAD

**Issue by inureyes** [lablup/mlxcel#1717]
## Problem

`text_only_forward_produces_finite_logits` in `tests/granite4_vision_parity.rs` fails sporadically at the assertion on `tests/granite4_vision_parity.rs:100`, against `granite-4.0-3b-vision-4bit` (2.8 GB, detected as `ModelType::Granite4VisionVLM`). The failure message now reports the value and its class, which is how the following distinction was established: it prints `nan=true, inf=false` on the max of the last logit row, so this is a NaN and not an overflow to infinity.

## Frequency

Sporadic and reproducible in isolation, not only under the full workspace gate. Measured 1 failure in 5 runs, then 1 in 8, then 2 in 3 consecutive full-gate runs. Each run is a fresh process and the failing call is the first forward after `load_model`, so there is no ordering or accumulation dependency.

## Scope

The two sibling tests with the same name and shape, `tests/smolvlm_parity.rs` and `tests/molmo_parity.rs`, passed 6 of 6 each, so this looks specific to the Granite 4 Vision path rather than to shared VLM test scaffolding.

This is pre-existing. It does not involve the Gemma3n dtype policy, the nemotron_nas rope work, or the sampling tolerance changed alongside it.

## What to check

NaN rather than infinity points away from overflow and toward a buffer that is read before it is fully written, and a sporadic fault on a fresh process with no ordering dependency fits the same family as #1712. Please inspect the first forward's intermediate tensors for an uninitialized read, starting with the KV cache and the attention mask allocation.

Related: #997 (open tracker for this assertion) and #1211 (closed as its duplicate).

## Verification

```
cargo test --profile test-fast --features metal,accelerate -p mlxcel --test granite4_vision_parity text_only_forward_produces_finite_logits
```

A pass is 20 of 20 repeated runs reporting finite logits.

**Comment by inureyes** [comment]
Duplicate of #997, which has covered `tests/granite4_vision_parity.rs:100` since 2026-08-02 and also carries the `hunyuan_vl_parity` half. #1211 was already closed into it for the same reason, so this is the third filing and I should have searched before opening it.

The findings are moved to #997 rather than lost: the value is NaN and not an infinity, `smolvlm_parity` and `molmo_parity` pass 6 of 6 each as controls, and the failure reproduces in isolation rather than only under the whole suite, which contradicts a premise stated in #997's own evidence table.

Closing in favour of #997.
